src/Chainweb/Pact/Backend/ChainwebPactDb.hs

```diff
@@ -427,194 +404,165 @@ createBlockHistoryTable =
-        exec_ db $
-            "CREATE TABLE IF NOT EXISTS "
-            <> Utf8 (toS (flip T.snoc ']' $ T.cons '[' $ (asString name)))
-            <> " (rowkey TEXT\
-               \, blockheight UNSIGNED BIGINT\
-               \, version UNSIGNED BIGINT\
-               \, txid UNSIGNED BIGINT\
-               \, rowdata BLOB \
-               \, UNIQUE(blockheight, rowkey, txid));"
-    versionedTablesInsert name b
-
-handleVersion :: BlockHeight -> ParentHash -> BlockHandler SQLiteEnv TxId
-handleVersion bRestore hsh = do
+createTableCreationTable :: BlockHandler SQLiteEnv ()
+createTableCreationTable =
+    callDb "createTableCreationTable" $ \db -> exec_ db
+      "CREATE TABLE IF NOT EXISTS VersionedTableCreation\
+      \(tablename TEXT NOT NULL\
+      \, createBlockheight UNSIGNED BIGINT NOT NULL\
+      \, CONSTRAINT creation_unique UNIQUE(tablename, createBlockheight));"
+
+createTableMutationTable :: BlockHandler SQLiteEnv ()
+createTableMutationTable =
+    callDb "createTableMutationTable" $ \db -> do
+      exec_ db "CREATE TABLE IF NOT EXISTS VersionedTableMutation\
+               \(tablename TEXT NOT NULL\
+               \, blockheight UNSIGNED BIGINT NOT NULL\
+               \, CONSTRAINT mutation_unique UNIQUE(tablename,blockheight));"
+      exec_ db "CREATE INDEX IF NOT EXISTS mutation_bh ON VersionedTableMutation(blockheight);"
+
+createUserTable :: TableName -> BlockHeight -> BlockHandler SQLiteEnv ()
+createUserTable name bh =
+    callDb "createUserTable" $ \db -> do
+      createVersionedTable tablename indexname db
+      exec' db insertstmt insertargs
+  where
+    insertstmt = "INSERT OR IGNORE INTO VersionedTableCreation VALUES (?,?)"
+    insertargs =  [SText tablename, SInt (fromIntegral bh)]
+    tablename = Utf8 (toS $ asString name)
+    indexname = Utf8 (toS $ asString name <> "_bh")
+
+createVersionedTable :: Utf8 -> Utf8 -> Database -> IO ()
+createVersionedTable tablename indexname db = do
+    exec_ db createtablestmt
+    exec_ db indexcreationstmt
+  where
+    createtablestmt =
+      "CREATE TABLE IF NOT EXISTS["
+        <> tablename
+        <> "] (rowkey TEXT\
+             \, blockheight UNSIGNED BIGINT NOT NULL\
+             \, txid UNSIGNED BIGINT NOT NULL\
+             \, rowdata BLOB NOT NULL\
+             \, UNIQUE (blockheight, rowkey, txid));"
+    indexcreationstmt = "CREATE INDEX IF NOT EXISTS [" <> indexname  <> "] ON [" <> tablename <> "](rowkey, txid);"
+
+handlePossibleRewind :: BlockHeight -> ParentHash -> BlockHandler SQLiteEnv TxId
+handlePossibleRewind bRestore hsh = do
     bCurrent <- getBCurrent
-    vCurrent <- getVCurrent
     checkHistoryInvariant
-
     case compare bRestore (bCurrent + 1) of
-        GT -> internalError "handleVersion: Block_Restore invariant violation!"
-        EQ -> newChildBlock bCurrent vCurrent
-        LT -> rewindBlock vCurrent
-
+      GT -> internalError "handlePossibleRewind: Block_Restore invariant violation!"
```

Comment:
e.g. you broke the indent here (it was correct before)

Reply:
done & done